### PR TITLE
Fix airloop topology crash

### PR DESCRIPTION
**Pull request by mjwitte**
Pull request overview
---------------------
 - Fixes crash in airloop topology report.
 - Defect test file: 
[IECC-state-current-4C_OfficeSmall_STD2021_WA-Seattle-WA_DM.idf.txt](https://github.com/user-attachments/files/16911276/IECC-state-current-4C_OfficeSmall_STD2021_WA-Seattle-WA_DM.idf.txt)

### Pull Request Author
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [x] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [x] Label the PR with at least one of: Defect, Refactoring, NewFeature, Performance, and/or DoNoPublish
 - [ ] Pull requests that impact EnergyPlus code must also include unit tests to cover enhancement or defect repair
 - [ ] Author should provide a "walkthrough" of relevant code changes using a GitHub code review comment process
 - [ ] If any diffs are expected, author must demonstrate they are justified using plots and descriptions
 - [ ] If changes fix a defect, the fix should be demonstrated in plots and descriptions
 - [ ] If any defect files are updated to a more recent version, upload new versions here or on DevSupport
 - [ ] If IDD requires transition, transition source, rules, ExpandObjects, and IDFs must be updated, and add IDDChange label
 - [ ] If structural output changes, add to output rules file and add OutputChange label
 - [ ] If adding/removing any LaTeX docs or figures, update that document's CMakeLists file dependencies

### Reviewer
This will not be exhaustively relevant to every PR.
 - [ ] Perform a Code Review on GitHub
 - [ ] If branch is behind develop, merge develop and build locally to check for side effects of the merge
 - [ ] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] If feature, test running new feature, try creative ways to break it
 - [ ] CI status: all green or justified
 - [ ] Check that performance is not impacted (CI Linux results include performance check)
 - [ ] Run Unit Test(s) locally
 - [ ] Check any new function arguments for performance impacts
 - [ ] Verify IDF naming conventions and styles, memos and notes and defaults
 - [ ] If new idf included, locally check the err file and other outputs

### CODE REVIEW

**Comment by mjwitte** [comment]
This is ready for review. Defect file crashes with develop, runs to completion here.

Audit diffs for any file that has more than one airloop serving the same zone, which results in huge changes in the table HVAC topology report. So, why no table diffs indicated? Hmmm.

**Comment by mjwitte** [comment]
Anyway, here's an example of the table output for 5ZoneAirCooledWithDOASAirLoop

With develop, both the regular and DOAS terminal units show as being on both airloops.
image

image


With this fix, they're each on their proper air loop.
image

image

File: src/EnergyPlus/DataAirSystems.hh [hunk 1]
```diff
@@ -190,7 +190,7 @@ namespace DataAirSystems {
         AirLoopMixerData Mixer;            // Data for mixer (if any)
         Array1D_bool ControlConverged;     // Convergence Parameter for controllers
         int NumOutletBranches = 0;
-        std::array<int, 3> OutletBranchNum = {0}; // branch numbers of system outlets
+        std::array<int, 3> OutletBranchNum = {0}; // airloop branch numbers of system outlets (not the actual branch index)
```
Comment (Myoldmopar) [inline src/EnergyPlus/DataAirSystems.hh:193]: Good clarification

**Review by Myoldmopar — APPROVED** [review]
Straightforward fix, this is good to go.

**Comment by Myoldmopar** [comment]
No need to wait on this fix, thanks @mjwitte

**Pull request merged**